lib/project_types/script/layers/infrastructure/script_service.rb

```diff
@@ -63,73 +66,31 @@ def push(
           end
         end
 
-        def get_app_scripts(api_key:, extension_point_type:)
+        def get_app_scripts(extension_point_type:)
           query_name = "get_app_scripts"
           variables = { appKey: api_key, extensionPointName: extension_point_type.upcase }
-          response = MakeRequest.new(ctx, api_key).call(
-            query_name: query_name,
-            variables: variables
-          )
+          response = make_request(query_name: query_name, variables: variables)
           response["data"]["appScripts"]
         end
 
-        class MakeRequest
-          attr_reader :ctx
-
-          def initialize(ctx, api_key)
-            @client = ApiClients.default_client(ctx, api_key)
-          end
-
-          def call(query_name:, variables: {})
-            resp = @client.query(query_name, variables: variables)
-            raise_if_graphql_failed(resp)
-            resp
-          end
+        def generate_module_upload_url
+          query_name = "module_upload_url_generate"
+          variables = {}
+          response = make_request(query_name: query_name, variables: variables)
+          user_errors = response["data"]["moduleUploadUrlGenerate"]["userErrors"]
 
-          def raise_if_graphql_failed(response)
-            raise Errors::EmptyResponseError if response.nil?
-            raise Errors::GraphqlError, response["errors"] if response.key?("errors")
-          end
+          raise Errors::GraphqlError, user_errors if user_errors.any?
```

Comment:
I am not sure if it's appropriate to raise `GraphqlError`?  What kind of error could actually happen?  I see this `FailToGenerate` error from SS.  Should that result in `GraphqlError` here?

Reply:
Keep in mind most of this is structural changes (this is just the body of the `UploadScript#call` method). I have not changed the underlying functionality, or at least have avoided it as much as possible. I think this should be addressed in a subsequent PR.